meson.build

```diff
@@ -0,0 +1,455 @@
+      dynarec_src += [['new_dynarec/arm', ['linkage_arm.S', 'arm_cpu_features.c']]]
+    else
+      error('New Dynarec is only supported on 32 bit x86 and 32 bit armel')
+    endif
+  else
+    # Hacktarux dynarec
+    if not 'x86 x86_64'.contains(host_cpu)
+      error('Unsupported architecture for dynarec: ' + host_cpu)
+    endif
+    dynarec_src = [[host_cpu, ['assemble.c', 'dynarec.c', 'regcache.c']]]
+    dynarec_asm = [[host_cpu, ['dyna_start.asm']]]
+  endif
+
+else
+  # dynarec is disabled
+  cflags += '-DNO_ASM'
+
+  dynarec_src = [['', ['empty_dynarec.c']]]
+  dynarec_asm = []
+endif
+r4300_dir = 'src/device/r4300'
+foreach x : dynarec_src
+  foreach f : x[1]
+    src += join_paths(r4300_dir, x[0], f)
+  endforeach
+endforeach
+foreach x : dynarec_asm
+  foreach f : x[1]
+    src_nasm += join_paths(r4300_dir, x[0], f)
+  endforeach
+endforeach
+
+
+# Add GLES support
+if with_gles
+  cflags += ['-DUSE_GLES']
+  libgles = cc.find_library('GLESv2')
+
+  # Setup VideoCore support
+  if with_vc
+    vc_dir = '/opt/vc/'
+    vc_lib_dir = join_paths(vc_dir, 'lib')
+    vc_inc_dir = join_paths(vc_dir, 'include')
+
+    inc += include_directories([
+      vc_inc_dir,
+      join_paths(vc_inc_dir, 'interface/vcos/pthreads'),
+      join_paths(vc_inc_dir, 'vmcs_host/linux')])
+
+    foreach x : ['brcmEGL', 'bcm_host', 'vcos', 'vchiq_arm']
+      extra_dep += cc.find_library(x, dirs : vc_lib_dir)
+    endforeach
+    # override libgles implementation
+    libgles = cc.find_library('brcmGLESv2', dirs : vc_lib_dir)
+  endif
+
+  extra_dep += libgles
+endif
+
+# Assembler step
+if src_nasm.length() > 0
+  # Compile asm_defines.c to extract compiler/os-specific offsets
+  # Use the same compilation flags as the other files of the project
+  # except for LTO which must be disabled for effective creation of required symbols
+  asm_defines_lib = static_library('asm_defines',
+    'src/asm_defines/asm_defines.c',
+    include_directories : inc,
+    c_args : cflags,
+    override_options : 'b_lto=false')
+
+  # Generate headers using an awk parser directly on the asm_defines_lib binary
+  awk = find_program('awk', join_paths(m64pwindeppath, 'gawk-3.1.6-1/bin/gawk.exe'))
+  asm_defines_headers = custom_target('asm_defines_headers',
+      input : ['tools/gen_asm_defines.awk', asm_defines_lib],
+      output : ['asm_defines_nasm.h', 'asm_defines_gas.h'],
+      command : [awk, '-v', 'dest_dir=@OUTDIR@', '-f', '@INPUT0@', '@INPUT1@'])
+
+  asflags = []
+
+  # forward PIC mode to assembler files
+  if not get_option('b_staticpic')
+    asflags += ['-d', 'PIC']
+  endif
+
+  # Select appropriate format
+  if 'linux bsd'.contains(host_system)
+    nasm_fmt = 'elf' + (host_cpu.endswith('64') ? '64' : '')
+  elif host_system == 'darwin'
+    nasm_fmt = 'macho' + (host_cpu.endswidth('64') ? '64' : '')
+  elif host_system == 'windows'
+    nasm_fmt = 'win' + (host_cpu.endswith('64') ? '64' : '32')
+  else
+    error('Unsupported system')
```

Comment:
A warning would be more friendly than an error here. Is nasm a hard dependency?

Reply:
nasm is a hard dependency (provided that you enable dynarecs), so erroring is the right thing.